tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -4776,5 +4781,22 @@ void DeclResultIdMapper::storeOutStageVarsToStorage(
   }
 }
 
+void DeclResultIdMapper::registerCapabilitiesAndExtensionsForType(
+    const TypedefType *type) {
+  for (const auto *decl : typeAliasesWithAttributes) {
+    if (type == decl->getTypeForDecl()) {
+      assert(decl->hasAttrs());
```

Comment:
Is this assert useful?
(iterators should be fine if no attr are presents, and its presence doesn't guarantee we have at least one of them no?)

Reply:
I think I had a different implementation earlier that would assert if the decl has no attributes, so I added this assert to make sure nothing was added to typeAliasesWithAttributes that did not have any attributes.

However, that is not true with this implementation. I've removed it.